Automattic/vip-go-mu-plugins
# Add a Prometheus collector to count the number of SQL queries with multi datasets

## Description

This PR adds a Prometheus collector to count (and identify) the queries that are performing operations across multiple DB datasets.

For now, it only runs when the feature toggle `VIP_POTENTIAL_MULTI_DATASET_QUERIES_COLLECTOR_ENABLED` is `true`. 

## Changelog Description

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:


## Pre-deploy checklist


## Steps to Test

1. Check out PR.
2. Perform a SQL query across multiple datasets
3. Observe the Prometheus metrics
